network-primitives/scripts/create_domain_node_compose_file.sh

```diff
@@ -0,0 +1,168 @@
+      caddy_1.handle_path_0: /http
+      caddy_1.handle_path_0.reverse_proxy: "{{upstreams 8933}}"
+      caddy_1.handle_path_1: /ws
+      caddy_1.handle_path_1.reverse_proxy: "{{upstreams 8944}}"
+      caddy_2: \${DOMAIN_PREFIX}-\${NODE_ID}.\${DOMAIN_LABEL}.\${NETWORK_NAME}.subspace.network
+      caddy_2.handle_path_0: /http
+      caddy_2.handle_path_0.reverse_proxy: "{{upstreams 7933}}"
+      caddy_2.handle_path_1: /ws
+      caddy_2.handle_path_1.reverse_proxy: "{{upstreams 7944}}"
+    command: [
+      "--chain", \$NETWORK_NAME,
+      "--base-path", "/var/subspace",
+      "--execution", "wasm",
+      "--state-pruning", "archive",
+      "--blocks-pruning", "archive",
+      "--listen-addr", "/ip4/0.0.0.0/tcp/30333",
+      "--no-private-ipv4",
+      "--dsn-disable-private-ips",
+      "--piece-cache-size", \$PIECE_CACHE_SIZE,
+      "--node-key", \$NODE_KEY,
+      "--rpc-cors", "all",
+      "--rpc-port", "9933",
+      "--ws-port", "9944",
+      "--ws-external",
+      "--in-peers", "500",
+      "--out-peers", "250",
+      "--in-peers-light", "500",
+      "--ws-max-connections", "10000",
+EOF
+
+reserved_only=${1}
+node_count=${2}
+current_node=${3}
+bootstrap_node_count=${4}
+enable_domains=${5}
+domain_id=${6}
+
+for (( i = 0; i < node_count; i++ )); do
+  if [ "${current_node}" != "${i}" ]; then
+    addr=$(sed -nr "s/NODE_${i}_MULTI_ADDR=//p" /subspace/node_keys.txt)
+    echo "      \"--reserved-nodes\", \"${addr}\"," >> /subspace/docker-compose.yml
+    echo "      \"--bootnodes\", \"${addr}\"," >> /subspace/docker-compose.yml
+  fi
+done
+
+for (( i = 0; i < bootstrap_node_count; i++ )); do
+  addr=$(sed -nr "s/NODE_${i}_MULTI_ADDR=//p" /subspace/bootstrap_node_keys.txt)
+  echo "      \"--reserved-nodes\", \"${addr}\"," >> /subspace/docker-compose.yml
+  echo "      \"--bootnodes\", \"${addr}\"," >> /subspace/docker-compose.yml
+done
+
+for (( i = 0; i < dsn_bootstrap_node_count; i++ )); do
+  dsn_addr=$(sed -nr "s/NODE_${i}_MULTI_ADDR=//p" /subspace/dsn_bootstrap_node_keys.txt)
+  echo "      \"--dsn-bootstrap-nodes\", \"${dsn_addr}\"," >> /subspace/docker-compose.yml
+done
+
+
+if [ "${reserved_only}" == true ]; then
+    echo "      \"--reserved-only\"," >> /subspace/docker-compose.yml
+fi
+
+if [ "${enable_domains}" == true ]; then
+    {
+    # system domain
+      echo '      "--",'
+      echo "      \"--chain=\$NETWORK_NAME\","
+      echo '      "--validator",'
+      echo '      "--state-pruning", "archive",'
+      echo '      "--blocks-pruning", "archive",'
+      echo '      "--base-path", "/var/subspace/system_domain",'
+      echo '      "--keystore-path", "/var/subspace/keystore",'
+      echo '      "--rpc-cors", "all",'
+      echo '      "--rpc-port", "8933",'
+      echo '      "--ws-port", "8944",'
+      echo '      "--no-private-ipv4",'
+      echo '      "--unsafe-ws-external",'
+      echo "      \"--relayer-id=\$RELAYER_SYSTEM_ID\","
+      if [ "${domain_id}" == 1 ]; then

```

Comment:
no, this is not the way. 
We made it configurable so we can avoid these if else. We can totally remove these IMO